### PR TITLE
Added doi field and improved dataset validation

### LINKED ISSUE #414: [Fronend] - Add a DOI to the dataset upload form
As a user, I want to be able to specify the DOI of the research paper I am referencing in my dataset

**Description** 
Backend should already support a DOI on the model, check up with Sean
Backend should already support a range of pages as a string.
Fix validation of the form to adjust required and not required fields
**Additional Notes**
some renaming should be done as part of this PR

**Tasks**
- [ ] Fix validation schema of the dataset form, to properly specify the required fields and remove required where it is not needed
- [ ] change the 'uploadDataset' route to just 'dataset'
- [ ] make the pages a string of type: 'number-number' || 'number' to support ranges of pages
- [ ] rename the files of the Dataset Upload View and DatasetUploadForm, into
- [ ] when a dataset is saved successfully display a Success message and redirect the user to the site.com/dataset/{returnedId}
- [ ] general improvements to form validations and form itself

**Acceptance Criteria**
a user can specify a doi and see their doi on created datasets

Priority High
Risks Low

### PR DESCRIPTION
**Pull request by ds161833**
image
removed unused packages, fixed some bugs

### CODE REVIEW

File: Client/package.json [hunk 1]
```diff
@@ -19,19 +19,11 @@
...
-    "enzyme-matchers": "^7.1.2",
-    "equation-parser": "^0.3.1",
-    "equation-resolver": "^0.3.2",
     "factory.ts": "^0.5.1",
     "faker": "^5.4.0",
     "formik": "^2.2.6",
-    "immutable": "^4.0.0-rc.12",
     "jest": "^24.9.0",
     "lodash": "^4.17.20",
-    "mocha": "^8.2.1",
```
Comment (TiraFire) [inline Client/package.json:34]: Thanks for removing these unused ones! we definitely needed the clean up

**Review by TiraFire — APPROVED** [review]
Looks good! much clearer to use than before and the use of the decorator pattern is great! Thanks for removing some unused code as well, cleans up our files. Keep up the amazing work ^^

File: Client/src/Components/DatasetUpload/MetaSection/MetaForm.tsx [hunk 2]
```diff
@@ -29,8 +29,9 @@ const getOptions = (options: any[]): any => {
 }
 
 export const MetaForm = (props: IProps) => {
+  console.log('meta rendered')
```
Comment (laithAwad) [inline Client/src/Components/DatasetUpload/MetaSection/MetaForm.tsx]: do you need this console log?

**Review by lposo — APPROVED** [review]
Nice implementation of the decorator pattern, good work!

File: Client/src/Components/DatasetUpload/MetaSection/MetaForm.tsx (hunk 2 above)
Comment (ds161833) [reply Client/src/Components/DatasetUpload/MetaSection/MetaForm.tsx]: nope they're gone

**Review by laithAwad — APPROVED** [review]
Great work! one small comment left

**Pull request merged**